MeshLib/Mesh.cpp

```diff
@@ -337,4 +338,58 @@ void scaleMeshPropertyVector(MeshLib::Mesh & mesh,
         v *= factor;
 }
 
+std::unique_ptr<MeshLib::Mesh> createMeshFromElementSelection(
+    std::string mesh_name, std::vector<MeshLib::Element*> const& elements)
+{
+    DBUG("Found %d elements in the mesh", elements.size());
+
+    // Store bulk element ids for each of the new elements.
+    std::vector<std::size_t> bulk_element_ids;
+    std::transform(begin(elements), end(elements),
+                   std::back_inserter(bulk_element_ids),
+                   [&](auto const& e) { return e->getID(); });
+
+    // original node pointers to newly created nodes.
+    std::unordered_map<const MeshLib::Node*, MeshLib::Node*> nodes_map;
+
+    for (auto& e : elements)
+    {
+        // For each node find a cloned node in map or create if there is none.
+        unsigned const n_nodes = e->getNumberOfNodes();
+        for (unsigned i = 0; i < n_nodes; ++i)
+        {
+            const MeshLib::Node* n = e->getNode(i);
+            auto it = nodes_map.find(n);
+            if (it == nodes_map.end())
+            {
+                nodes_map[n] = new MeshLib::Node(*n);
+            }
+
+            e->setNode(i, nodes_map[n]);
+        }
+    }
+
+    // Copy the unique nodes pointers.
+    std::vector<MeshLib::Node*> element_nodes;
+    std::transform(begin(nodes_map), end(nodes_map),
+                   std::back_inserter(element_nodes),
+                   [&](auto const& pair) { return pair.second; });
+
+    // Store bulk node ids for each of the new nodes.
+    std::vector<std::size_t> bulk_node_ids;
+    std::transform(begin(nodes_map), end(nodes_map),
+                   std::back_inserter(bulk_node_ids),
+                   [&](auto const& pair) { return pair.first->getID(); });
+
+    auto mesh =
+        std::make_unique<MeshLib::Mesh>(mesh_name, element_nodes, elements);
```

Comment:
Shouldn't the element and nodes vectors be moved?

Reply:
Commented in function's docu and in the code.